<launch>

  <!-- <rosparam command="load" file="$(find amswarmx)/params/config_am_swarm.yaml"/> -->
  <!-- <node pkg="rosbag" type="record" name="rosbag_record" args="record -a -o $(find amswarmx)/data/bag/"/>    -->

  <arg name="world_frame_id" default="map"/> <!--  -->
  <arg name="world_file_name" default="room.bt"/> <!-- Octomap file name, it must be bt file -->
  <arg name="world_resolution" default="0.1"/> 
  <arg name="display_freq" default="10"/> 
  <arg name="display_sim" default="true" />

  <node pkg="amswarmx" type="swarm_am_nav" name="swarm_am_nav" output="screen">
    <param name="world/frame_id" value="$(arg world_frame_id)" />
    <param name="world/file_name" value="$(arg world_file_name)" />
    <param name="world/resolution" value="$(arg world_resolution)" />
    <param name="rviz/display_freq" value="$(arg display_freq)"/> 
    <param name="rviz/display_sim" value="$(arg display_sim)" />
  </node>

  
  <node pkg="rviz" type="rviz" name="rviz" args="-d $(find amswarmx)/launch/rviz_config/ortho_config.rviz" if="$(arg display_sim)"/>

  <node pkg="octomap_server" type="octomap_server_node" name="octomap_server"
        args="$(find amswarmx)/world/$(arg world_file_name)">
    <param name="resolution"                 value="$(arg world_resolution)" />
    <param name="frame_id"                   value="$(arg world_frame_id)" type="string"/>
    <param name="sensor_model/max_range"     value="50.0" />
    <param name="height_map"                 value="false"/>
    <param name="color/r"                    value="0.2" />
    <param name="color/g"                    value="0.2" />
    <param name="color/b"                    value="0.2" />
    <param name="color/a"                    value="0.2" />
  </node>
</launch>
